src/net.cpp

```diff
@@ -2331,6 +2332,11 @@ bool CAddrDB::Read(CAddrMan& addr)
 
     CDataStream ssPeers(vchData, SER_DISK, CLIENT_VERSION);
 
+    return Read(addr, ssPeers, hashIn);
+}
+
+bool CAddrDB::Read(CAddrMan& addr, CDataStream& ssPeers, uint256& hashIn)
+{
     // verify stored checksum matches input data
     uint256 hashTmp = Hash(ssPeers.begin(), ssPeers.end());
```

Comment:
Ok, yes, that makes sense.

> How interested are you in a system for mocking out filesystem objects rather than having to do everything at the stream level?

Depends on the (non-test) code impact. I'm not entirely sure that is necessary - e.g. you could create a temporary file and make it read that.